### ISSUE TITLE
Refactor code in `SemanticsDeclBodyVisitor::visitAggTypeDecl`

### ISSUE THREAD

**Issue by kaizhangNV** [shader-slang/slang#5135]
Backgroud:

This sub-task is split from https://github.com/shader-slang/slang/issues/3406.

I wrap up Arial's PR https://github.com/shader-slang/slang/pull/4854 to a new PR https://github.com/shader-slang/slang/pull/5107 with some minor issues solved.

However, there still be lots of comments could potential involve large change on this PR.

We should not make one PR has infinitely long conversion and unlimited problems solved, so this task is create to only spot this issue.

Here is the comment:
https://github.com/shader-slang/slang/pull/5107/files/2b6f7a26688b60a74fde979fbfff28a24a493e5c#r1769320448

TLDR: The implementation in SemanticsDeclBodyVisitor::visitAggTypeDecl is lack of readability, it's difficult for anyone to really understand the logic. So we need to refactor the code by splitting the logic into smaller function.
